### ISSUE TITLE
How to reproduce the results of fully supervised learning with only 40 labels

### ISSUE THREAD

**Issue by ylsung** [google-research/fixmatch#61]
Table 9 shows that you can achieve about 36% accuracy in a supervised manner by only using 40 labels. However, I didn't see the codes to reproduce it, can you share the command/codes you use? Plus, may I know the hyperparameters for the experiment because it is lacking in the paper. Thanks.

**Comment by carlini** [comment]
The code to reproduce is here fully_supervised/fs_baseline.py

We may have done a quick sweep of the weight decay (probably increasing it) but don't think we tuned anything else to adjust.

**Comment by ylsung** [comment]
I see. But fully_supervised/fs_baseline.py seems for supervised learning with all the labeled data (50000 for cifar10), I am thinking about how to adjust it for only 40 labels.

**Comment by ylsung** [comment]
I fixed it by combining the fully_supervised/lib/data.py and libml/data.py, thanks for the help. 

As for the hyperparameters, I am wondering you train 1 << 16 << 10 (67108864) iterations on 40 labeled data? It seems too much for such small data.

**Comment by carlini** [comment]
I think that we just did that, but training doesn't collapse, it just stops improving. See here for the full 50k run:

https://github.com/google-research/fixmatch/blob/d4985a158065947dba803e626ee9a6721709c570/fully_supervised/runs/all.sh#L18

I expect if you started one job you'll see it plateau at some point and it should be approximately what we report in the paper. Once you know how many steps that is just set the number of iterations to that.
